<a href="https://colab.research.google.com/github/sebastianacebedo/CPE-103-OOP-1-A/blob/main/oopfa1_Acebedo_lab8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [15]:
# distance is a class. Distance is measured in terms of feet and inches
class Distance:
    def __init__(self, f, i):
        self.feet = f
        self.inches = i

    # Overloading the binary operator > to compare two distances
    def __gt__(self, d):
        if self.feet > d.feet:
            return True
        elif self.feet == d.feet and self.inches > d.inches:
            return True
        else:
            return False

    # Overloading the binary operator + to add two distances
    def __add__(self, d):
        i = self.inches + d.inches
        f = self.feet + d.feet
        if i >= 12:
            i -= 12
            f += 1
        return Distance(f, i)

    # Displaying the distance
    def show(self):
        print("Feet =", self.feet, "Inches =", self.inches)

# Input for the distances
a, b = input("Enter feet and inches of distance1: ").split()
a, b = [int(a), int(b)]

c, d = input("Enter feet and inches of distance2: ").split()
c, d = [int(c), int(d)]

# Create two distance objects
d1 = Distance(a, b)
d2 = Distance(c, d)

# Compare the distances
if d1 > d2:
    print("Distance1 is greater than Distance2")
else:
    print("Distance2 is greater or equal to Distance1")

# Add the two distances
d3 = d1 + d2
print("Sum of the two distances is:")
d3.show()

Enter feet and inches of distance1: 5 10
Enter feet and inches of distance2: 2 3
Distance1 is greater than Distance2
Sum of the two distances is:
Feet = 8 Inches = 1


In [17]:
class RegularPolygon:
    def __init__(self, side):
        self._side = side

class Square(RegularPolygon):
    def area(self):
        return self._side * self._side

class EquilateralTriangle(RegularPolygon):
    def area(self):
        return self._side * self._side * 0.433

obj1 = Square(4)
obj2 = EquilateralTriangle(3)

print(obj1.area())
print(obj2.area())

16
3.897


#SUPPLEMENTARY ACTIVITY

In [21]:
class RegularPolygon:
    #for all regular polygons
    def __init__(self, side):
        self._side = side

class Square(RegularPolygon):
    # Square has 4 equal sides
    def area(self):
        return round(self._side * self._side, 4)

class EquilateralTriangle(RegularPolygon):
    # Triangle has 3 equal sides, area calculated using 0.433 * side^2
    def area(self):
        return round(self._side * self._side * 0.433, 4)

class Pentagon(RegularPolygon):
    # Pentagon has 5 equal sides, area calculated using a special formula
    def area(self):
        return round((1/4) * (5 * (5 + 2 * (5 ** 0.5))) ** 0.5 * self._side ** 2, 4)

class Hexagon(RegularPolygon):
    # Hexagon has 6 equal sides, area calculated using a specific formula
    def area(self):
        return round((3 * (3 ** 0.5) / 2) * self._side ** 2, 4)

class Octagon(RegularPolygon):
    # Octagon has 8 equal sides, area calculated using a derived formula
    def area(self):
        return round(2 * (1 + (2 ** 0.5)) * self._side ** 2, 4)

# Creating objects for each shape with given side lengths
obj1 = Square(4)
obj2 = EquilateralTriangle(3)
obj3 = Pentagon(5)
obj4 = Hexagon(6)
obj5 = Octagon(7)

# Printing calculated
print("Square Area: %.2f" % obj1.area())
print("Equilateral Triangle Area: %.2f" % obj2.area())
print("Pentagon Area: %.2f" % obj3.area())
print("Hexagon Area: %.2f" % obj4.area())
print("Octagon Area: %.2f" % obj5.area())


Square Area: 16
Equilateral Triangle Area: 3.897
Pentagon Area: 43.01
Hexagon Area: 93.53
Octagon Area: 236.59


Square (Side = 4)
The area is 16.00, calculated using the formula side × side (4 × 4).

Equilateral Triangle (Side = 3)
The area is 3.90, calculated using the formula 0.433 × side² (0.433 × 3²).

Pentagon (Side = 5)
The area is 43.01, calculated using a specific pentagon area formula.

Hexagon (Side = 6)
The area is 93.53, calculated using the formula (3√3 / 2) × side².

Octagon (Side = 7)
The area is 118.74, calculated using the formula 2 × (1 + √2) × side².





